src/test/coins_tests.cpp

```diff
@@ -53,9 +53,9 @@ class CCoinsViewTest : public CCoinsView
 
     uint256 GetBestBlock() const override { return hashBestBlock_; }
 
-    bool BatchWrite(CCoinsMap& mapCoins, const uint256& hashBlock) override
+    bool BatchWrite(CCoinsMap& mapCoins, const uint256& hashBlock, bool erase = true) override
     {
-        for (CCoinsMap::iterator it = mapCoins.begin(); it != mapCoins.end(); ) {
+        for (CCoinsMap::iterator it = mapCoins.begin(); it != mapCoins.end(); it = erase ? mapCoins.erase(it) : ++it) {
```

Comment:
In the third commit  2c3cbd6c007a588e667, it looks like https://github.com/bitcoin/bitcoin/pull/17487#discussion_r1082702771 would apply here as well.

```suggestion
        for (CCoinsMap::iterator it = mapCoins.begin(); it != mapCoins.end(); it = erase ? mapCoins.erase(it) : std::next(it)) {
```